extensions/db/src/main/java/org/mule/extension/db/internal/operation/BaseDbOperations.java

```diff
@@ -97,6 +102,22 @@ protected void validateQueryType(QueryTemplate queryTemplate, List<QueryType> va
     }
   }
 
+
+  protected void validateNoParameterTypeIsUnused(Query query, List<ParameterType> parameterTypes) {
+    Set<String> params = query.getQueryTemplate().getParams().stream().map(QueryParam::getName).collect(Collectors.toSet());
+    Set<ParameterType> unusedTypes =
+        parameterTypes.stream().filter(type -> !params.contains(type.getKey())).collect(Collectors.toSet());
+
+    if (!unusedTypes.isEmpty()) {
+      throw new IllegalArgumentException(format("Query parameters [%s] were unused during query resolution, please remove them",
+                                                Joiner.on(", ")
+                                                    .join(unusedTypes.stream()
+                                                        .map(type -> format("('%s' of %s type)", type.getKey(),
```

Comment:
no need to include the type. This will just make the error message difficult to read